### PR TITLE
Json args

### PR DESCRIPTION
**Pull request by weeryan17**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] Your code follows the same code style as the rest of the project
- [x] Tested changes

Switch command line args over to the json config

PS) Based off my warn branch so merge that first.

### CODE REVIEW

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 1]
```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
...
+            FlareBot.LOGGER.error("Unable to create config.json!", e);
+            System.exit(1);
+        }
+
+        List<String> required = new ArrayList<>();
+        required.add("bot.token");
+        required.add("bot.statusHook");
+        required.add("cassandra.username");
+        required.add("cassandra.password");
+        required.add("mysc.yt");
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: *misc

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 2]
```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
...
-        CommandLineArguments.parse(args);
+        JSONConfig config = null;
+        try {
+            File file = new File("config.json");
+            if (!file.exists())
+                file.createNewFile();
+            try {
+                config = new JSONConfig("config.json");
+            } catch (NullPointerException e) {
+                FlareBot.LOGGER.error("Invalid JSON!", e);
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: Just use LOGGER, you're in the FlareBot class

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 3]
```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
...
+                System.exit(1);
+            }
+        } catch (IOException e) {
+            FlareBot.LOGGER.error("Unable to create config.json!", e);
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: ^

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 4]
```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
...
+        FlareBot.pass = config.getString("cassandra.password").get();
 
-        FlareBot.secret = CommandLineArguments.SECRET.getValue();
-        FlareBot.dBotsAuth = CommandLineArguments.DBOTS.getValue();
-        FlareBot.webSecret = CommandLineArguments.WEBSECRET.getValue();
-        if (CommandLineArguments.DEBUG.isSet()) {
-            ((ch.qos.logback.classic.Logger) LoggerFactory
-                    .getLogger(ch.qos.logback.classic.Logger.ROOT_LOGGER_NAME))
-                    .setLevel(Level.DEBUG);
+        if (config.getArray("cassandra.nodes").isPresent()) {
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: You shouldn't handle Cassandra here, you can use the config in the CassandraController init. There is no need to use it here and do this. You are just making variables that will be used once and just exposing more info than is needed.

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 5]
```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
...
+                JsonElement em = it.next();
+                if (em.getAsString() != null) {
+                    ips.add(em.getAsString());
+                }
+            }
+            FlareBot.nodes = ips;
+        }
+
+        new CassandraController().init();
+        if (config.getString("mysc.hook").isPresent()) {
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: *misc

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 6]
```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
...
+            FlareBot.secret = config.getString("mysc.hook").get();
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: *misc

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 7]
```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
...
+        }
+        if (config.getString("botlists.discordBots").isPresent()) {
+            FlareBot.dBotsAuth = config.getString("botlists.discordBots").get();
+        }
+        if (config.getString("mysc.web").isPresent()) {
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: *misc

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 8]
```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
...
+            FlareBot.webSecret = config.getString("mysc.web").get();
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: *misc

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 9]
```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
...
+        }
+        FlareBot.statusHook = config.getString("bot.statusHook").get();
+        if (config.getString("botlists.botlist").isPresent()) {
+            FlareBot.botListAuth = config.getString("botlists.botlist").get();
+        }
+        FlareBot.youtubeApi = config.getString("mysc.yt").get();
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: *misc

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 10]
```diff
@@ -226,6 +298,18 @@ public static OkHttpClient getOkHttpClient() {
         return client;
     }
 
+    public static String getUser() {
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: None of these are needed or should be exposed at public level.

File: src/main/java/stream/flarebot/flarebot/FlareBot.java (hunk 4 above)
Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/FlareBot.java]: so should i open the config to everything else then?

**Pull request merged**